OpenRA.Game/Map/Map.cs

```diff
@@ -760,9 +802,8 @@ public void FixOpenAreas(Ruleset rules)
 
 		public byte GetTerrainIndex(CPos cell)
 		{
-			var uv = cell.ToMPos(this);
-			var custom = CustomTerrain[uv];
-			return custom != byte.MaxValue ? custom : cachedTileSet.Value.GetTerrainIndex(MapTiles.Value[uv]);
+			var custom = CustomTerrain[cell];
+			return custom != byte.MaxValue ? custom : cachedTileSet.Value.GetTerrainIndex(MapTiles.Value[cell]);
```

Comment:
Imagine you have an isometric map.

- GetTerrainIndex is called.
- ```CustomTerrain[cell]``` calls the CPos indexer, which calls ToMPos because we have an isometric map.
- Assume the result is ```byte.MaxValue```.
- The false path of the ternary statement fires, which includes ```MapTiles.Value[cell]```. This again calls the indexer which again calls ToMPos.